### DISCUSSION TITLE
git custom references

### DISCUSSION THREAD

**Discussion by iurly** [community/community#30507] [Other Feature Feedback, Questions, & Ideas]
Plain `git` allows you to create custom references, i.e. in addition to e.g. `refs/tags` for tags, `refs/heads` for branches, `refs/remotes...` for remote branches etc, using the following syntax:

```
git update-ref refs/myrefcategory/myref1 cac0ca
```

Being a fully-compatible git implementation, github.com allows you to push/pull such references from your local repo clones.
I *think* such references are accessible through github's custom REST API, but I could not find a way to list those references from github.com's web UI though.
Is this expected? Is there any workaround for this?

Thank you!

**Comment by acezxn** [comment]
I did not see any way you can list the references from github.com's web UI. However, github has its own API. To list the references with github's API, there are some ways to do that.

https://docs.github.com/en/rest/git/refs#list-matching-references
